### PR TITLE
Flink getting-started doc cleanup

### PR DESCRIPTION
**Pull request by xccui**
1. Refine the doc to improve readability;
2. Add some notes.

The sentences I felt uncertain with are marked with a series of question marks (????????????????).

### CODE REVIEW

**Comment by xccui** [comment]
Hi @stevenzwu, would you take some time to take a look? Thanks.

File: docs/content/docs/flink/flink-getting-started.md [hunk 1]
```diff
@@ -22,31 +22,31 @@ url: flink
...
 
 | Feature support                                             | Flink  | Notes                                                        |
 | ----------------------------------------------------------- | -----  | ------------------------------------------------------------ |
 | [SQL create catalog](#creating-catalogs-and-using-catalogs) | ✔️     |                                                              |
 | [SQL create database](#create-database)                     | ✔️     |                                                              |
 | [SQL create table](#create-table)                           | ✔️     |                                                              |
 | [SQL create table like](#create-table-like)                 | ✔️     |                                                              |
-| [SQL alter table](#alter-table)                             | ✔️     | Only support altering table properties, column and partition changes are not supported |
+| [SQL alter table](#alter-table)                             | ✔️     | Only support altering table properties. Changing column and partition are not supported. |
```
Comment (rdblue) [inline docs/content/docs/flink/flink-getting-started.md:34]: There is no need for punctuation here and this rephrase isn't making the point more clear. Can you please revert this change?

File: docs/content/docs/flink/flink-getting-started.md [hunk 2]
```diff
@@ -22,31 +22,31 @@ url: flink
 
 # Flink
 
-Apache Iceberg supports both [Apache Flink](https://flink.apache.org/)'s DataStream API and Table API. Currently,
-Iceberg integration for Apache Flink is available for Flink versions 1.12, 1.13, and 1.14. Previous versions of Iceberg also support Flink 1.11.
+Apache Iceberg supports both [Apache Flink](https://flink.apache.org/)'s DataStream API and Table API/SQL. Currently,
+Iceberg can with work Apache Flink versions 1.12, 1.13, and 1.14. Previous versions of Iceberg also support Flink 1.11.
```
Comment (rdblue) [inline docs/content/docs/flink/flink-getting-started.md:26]: Please revert this. This is no more clear than the previous statement.

File: docs/content/docs/flink/flink-getting-started.md [hunk 3]
```diff
@@ -33,20 +33,20 @@ Iceberg integration for Apache Flink is available for Flink versions 1.12, 1.13,
...
+| [SQL select](#querying-with-sql)                            | ✔️     | Support both streaming and batch mode.                       |
+| [SQL insert into](#insert-into)                             | ✔️ ️   | Support both streaming and batch mode.                       |
 | [SQL insert overwrite](#insert-overwrite)                   | ✔️ ️   |                                                              |
 | [DataStream read](#reading-with-datastream)                 | ✔️ ️   |                                                              |
 | [DataStream append](#appending-data)                        | ✔️ ️   |                                                              |
 | [DataStream overwrite](#overwrite-data)                     | ✔️ ️   |                                                              |
-| [Metadata tables](#inspecting-tables)                       | ️      | Support Java API but does not support Flink SQL              |
+| [Metadata tables](#inspecting-tables)                       | ️      | Only support in Java API but not in Flink SQL.               |
```
Comment (kbendick) [inline docs/content/docs/flink/flink-getting-started.md:42]: Nit:`Only supported with the Java API, not with Flink SQL.`

If you're updating the others, I'd update them as well to be more correctly phrased English sentences / phrases (so basically changing `only support` -> `only supports` or `only supported with` etc.

**Comment by xccui** [comment]
Hi @rdblue @kbendick, would you take another look when it's convenient? Thank you.

**Comment by xccui** [comment]
@rdblue @kbendick Sorry for bothering you again. I know you must have been busy working on some more urgent stuff. Just wonder if you still think this PR makes sense and would like me to keep it up to date with the latest change. Thanks!
